src/CommandProcessing/CommandProcessing.cpp

```diff
@@ -0,0 +1,197 @@
+        for (int i = 0; i < transitions.size(); i++) {
+            if (cmd->getCommand().compare(transitions[i]->getCommand()) == 0 
+                    && transitions[i]->getCurrentState()->getStateName() == p->getState()->getStateName()) {
+                        isValid = true;
+                        p->setState(transitions[i]->getNextState());
+                        cout << "this command is valid in the current game state.\n";
+                        string effect = "Player was moved to "+transitions[i]->getNextState()->getStateName();
+                        cmd->setEffect(effect);
+            }
+            if (isValid && i == transitions.size() - 1) {
+                cmd->setEffect("Invalid Command Error");
+                cout << "this command is NOT VALID in the current game state.\n";
+            }
+        }
+    }
+/*************************************************** FileCommandProcessorAdapter ***************************************************/
+/**
+ * Default Constructor
+ */
+    FileCommandProcessorAdapter::FileCommandProcessorAdapter(){}
+/**
+ * Destructor
+ */
+    FileCommandProcessorAdapter::~FileCommandProcessorAdapter(){}
+/**
+ * Assignment Operator
+ */
+    FileCommandProcessorAdapter& FileCommandProcessorAdapter::operator=(const FileCommandProcessorAdapter& other){
+        this->commands = other.commands;
+        return *this;
+    }
+/**
+ * Copy Constructor
+ */
+    FileCommandProcessorAdapter::FileCommandProcessorAdapter(const FileCommandProcessorAdapter& other){
+        this->commands = other.commands;
+    }
+/**
+ * overide Stream insertion operator
+ */
+    ostream& operator<<(ostream& out, FileCommandProcessorAdapter* cp){
+        //haven't decided what to printout so TODO!!!
+        return out;
+    }
+    /**
+     * reads commands line by line from an input file and saves the commands internally in a collection of Command objects
+    */
+    void FileCommandProcessorAdapter::readsaveCommand(){
+        string commandstr;
+        ifstream readCommands("Commands.txt");
+        while (getline (readCommands, commandstr)) {
+            Command* commandobj = new Command(commandstr);
+            commands.push_back(commandobj);
+        }
+    }
+/**
+ * public method for other objects such as the GameEngine or the Player
+ */
+    Command FileCommandProcessorAdapter::getCommandForOtherObjects(){
+        if (!commands.empty()) {
+            //TODO
+        }
+    }
+/**
+ * checks if the command is valid in the current game state
+ * let's say a player is in state issueorders but then gives a command that belong to executeorders
+ * this command should be rejected
+ * otherwise we update the state of the player
+*/
+    void FileCommandProcessorAdapter::validate(Command* cmd, Player* p){
+        bool isValid = false;
+        vector<Transition*> transitions = initializeGameTransitions();
+        for (int i = 0; i < transitions.size(); i++) {
```

Comment:
very similar to     `void CommandProcessor::validate(Command* cmd, Player* p)` , could this be a helper method instead?

Reply:
i thought about it too. i might make validate() and getCommandForOtherObjects() heleper methods for both FileCommandProcessorAdapter and CommandProcessor classes since they share them